src/infill/ZigzagConnectorProcessor.h

```diff
@@ -94,59 +80,130 @@ namespace cura
+     * \param use_endpieces Whether to include end pieces or not
+     * \param connected_endpieces Whether the end pieces should be connected with the rest part of the infill
+     * \param skip_some_zags; //!< Whether to skip some zags
+     * \param zag_skip_count; //!< Skip 1 zag in every N zags
      */
-    ZigzagConnectorProcessor(const PointMatrix& rotation_matrix, Polygons& result)
+    ZigzagConnectorProcessor(const PointMatrix& rotation_matrix, Polygons& result,
+                             bool use_endpieces, bool connected_endpieces,
+                             bool skip_some_zags = false, int zag_skip_count = 0)
     : rotation_matrix(rotation_matrix)
     , result(result)
+    , use_endpieces(use_endpieces)
+    , connected_endpieces(connected_endpieces)
+    , skip_some_zags(skip_some_zags)
+    , zag_skip_count(zag_skip_count)
+    , is_first_connector(true)
+    , first_connector_index(0)
+    , first_connector_direction(0)
+    , last_connector_index(0)
+    , last_connector_direction(0)
+    {}
+
+    virtual ~ZigzagConnectorProcessor()
     {}
-public:
 
     /*!
      * Handle the next vertex on the outer boundary.
      * \param vertex The vertex
      */
-    virtual void registerVertex(const Point& vertex) = 0;
-    
+    void registerVertex(const Point& vertex);
+
     /*!
      * Handle the next intersection between a scanline and the outer boundary.
      * 
      * \param intersection The intersection
-     * \param scanline_is_even Whether the scanline was even
+     * \param scanline_index Index of the current scanline
+     * \param direction Direction of this line segment
      */
-    virtual void registerScanlineSegmentIntersection(const Point& intersection, bool scanline_is_even) = 0;
-    
+    void registerScanlineSegmentIntersection(const Point& intersection, int scanline_index, int direction);
+
     /*!
      * Handle the end of a polygon and prepare for the next.
      * This function should reset all member variables.
      */
-    virtual void registerPolyFinished() = 0;
+    void registerPolyFinished();
+
+    /*!
+     * Reset the state so it can be used for processing another polygon.
+     */
+    void reset();
+
+protected:
+    const PointMatrix& rotation_matrix; //!< The rotation matrix used to enforce the infill angle
+    Polygons& result; //!< The result of the computation
+
+    bool use_endpieces; //!< Whether to include end pieces or not
+    bool connected_endpieces; //!< Whether the end pieces should be connected with the rest part of the infill
+    int skip_some_zags; //!< Whether to skip some zags
+    int zag_skip_count; //!< Skip 1 zag in every N zags
+
+    bool is_first_connector; //!< indicating whether we are still looking for the first connector or not
+    int first_connector_index; //!< scanline segment index of the first connector
+    int first_connector_direction; //!< direction of the first connector
```

Comment:
This variable doesn't need to be saved. See comment further on about computing the direction from the start adn end scnaline index